iceoryx_posh/include/iceoryx_posh/iceoryx_posh_types.hpp

```diff
@@ -94,6 +90,11 @@ constexpr uint32_t MAX_SUBSCRIBERS = build::IOX_MAX_SUBSCRIBERS;
 constexpr uint32_t MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY =
     build::IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY;
 constexpr uint32_t MAX_SUBSCRIBER_QUEUE_CAPACITY = MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY;
+// Introspection is using the following publisherPorts, which reduced the number of ports available for the user
+// 1x publisherPort mempool introspection
+// 1x publisherPort process introspection
+// 3x publisherPort port introspection
+constexpr uint32_t PUBLISHERS_RESERVED_FOR_INTROSPECTION = 5;
```

Comment:
nice.

The question is, should this be added to the specified number of ports or subtracted from them?

Reply:
Yes. I also thought about having an expression for ports available to users so it'll be explicit. I chose this way as it required less code changes now. @elBoberido What about adding this question to an introspection refactoring issue?

Reply:
sure

Reply:
Added to #414